src/phase0/state_transition/mod.rs

```diff
@@ -836,13 +833,117 @@ pub fn get_total_balance<
         PENDING_ATTESTATIONS_BOUND,
     >,
     indices: Vec<ValidatorIndex>,
-    context: &Context,
+    context: &Context
 ) -> Result<Gwei, Error> {
     let total_balance = indices
         .into_iter()
-        .try_fold(0usize, |acc, i| {
-            acc.checked_add(state.validators[i].effective_balance as usize)
+        .try_fold(0u64, |acc, i| {
+            acc.checked_add(state.validators[i].effective_balance)
         })
         .ok_or(Error::GetTotalBalance)?;
-    Ok(u64::max(total_balance as u64, EFFECTIVE_BALANCE_INCREMENT))
+    Ok(u64::max(total_balance as u64, context.effective_balance_increment))
 }
+
+pub fn get_total_active_balance<
+    const SLOTS_PER_HISTORICAL_ROOT: usize,
+    const HISTORICAL_ROOTS_LIMIT: usize,
+    const ETH1_DATA_VOTES_BOUND: usize,
+    const VALIDATOR_REGISTRY_LIMIT: usize,
+    const EPOCHS_PER_HISTORICAL_VECTOR: usize,
+    const EPOCHS_PER_SLASHINGS_VECTOR: usize,
+    const MAX_VALIDATORS_PER_COMMITTEE: usize,
+    const PENDING_ATTESTATIONS_BOUND: usize,
+>(
+    state: &BeaconState<
+        SLOTS_PER_HISTORICAL_ROOT,
+        HISTORICAL_ROOTS_LIMIT,
+        ETH1_DATA_VOTES_BOUND,
+        VALIDATOR_REGISTRY_LIMIT,
+        EPOCHS_PER_HISTORICAL_VECTOR,
+        EPOCHS_PER_SLASHINGS_VECTOR,
+        MAX_VALIDATORS_PER_COMMITTEE,
+        PENDING_ATTESTATIONS_BOUND,
+    >,
+    context: &Context,
+) -> Result<Gwei, Error> {
+    get_total_balance(state, get_active_validator_indices(state, get_current_epoch(state, context)), context)
+}
+
+pub fn get_indexed_attestation<
+    const SLOTS_PER_HISTORICAL_ROOT: usize,
+    const HISTORICAL_ROOTS_LIMIT: usize,
+    const ETH1_DATA_VOTES_BOUND: usize,
+    const VALIDATOR_REGISTRY_LIMIT: usize,
+    const EPOCHS_PER_HISTORICAL_VECTOR: usize,
+    const EPOCHS_PER_SLASHINGS_VECTOR: usize,
+    const MAX_VALIDATORS_PER_COMMITTEE: usize,
+    const PENDING_ATTESTATIONS_BOUND: usize,
+>(
+    state: BeaconState<
+        SLOTS_PER_HISTORICAL_ROOT,
+        HISTORICAL_ROOTS_LIMIT,
+        ETH1_DATA_VOTES_BOUND,
+        VALIDATOR_REGISTRY_LIMIT,
+        EPOCHS_PER_HISTORICAL_VECTOR,
+        EPOCHS_PER_SLASHINGS_VECTOR,
+        MAX_VALIDATORS_PER_COMMITTEE,
+        PENDING_ATTESTATIONS_BOUND,
+    >,
+    attestation: &mut Attestation<MAX_VALIDATORS_PER_COMMITTEE>,
```

Comment:
why is this `mut`?

we should not mutate the `attestation` to get the matching `IndexedAttestation`

Reply:
fixed